components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDAO.java

```diff
@@ -385,6 +414,156 @@ public OAuthAppDO getAppInformation(String consumerKey) throws
         return oauthApp;
     }
 
+    /**
+     * Get the OAuth consumer application for the given consumer key and access token.
+     *
+     * @param consumerKey   Consumer key of the OAuth application.
+     * @param accessTokenDO AccessTokenDO object.
+     * @return OAuthAppDO object.
+     * @throws InvalidOAuthClientException  Invalid OAuth client.
+     * @throws IdentityOAuth2Exception      Error while retrieving the OAuth application.
+     */
+    public OAuthAppDO getAppInformation(String consumerKey, AccessTokenDO accessTokenDO) throws
+            InvalidOAuthClientException, IdentityOAuth2Exception {
+
+        OAuthAppDO oauthApp = null;
+        String tokenId = accessTokenDO.getTokenId();
+        if (StringUtils.isBlank(tokenId)) {
+            throw new IdentityOAuth2Exception("Error while retrieving the application. Token id is empty.");
+        }
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            String sqlQuery = SQLQueries.OAuthAppDAOSQLQueries.GET_APP_INFO_FOR_TOKEN_ID_WITH_PKCE;
+
+            try (PreparedStatement prepStmt = connection.prepareStatement(sqlQuery)) {
+                prepStmt.setString(1, tokenId);
+
+                try (ResultSet rSet = prepStmt.executeQuery()) {
+                    /*
+                      We need to determine whether the result set has more than 1 row. Meaning, we found an
+                      application for the given consumer key. There can be situations where a user passed a key which
+                      doesn't yet have an associated application. We need to barf with a meaningful error message
+                      for this case.
+                    */
+                    boolean appExists = false;
+                    while (rSet.next()) {
+                        // There is at least one application associated with a given key.
+                        appExists = true;
+                        if (StringUtils.isNotBlank(rSet.getString(4))) {
```

Comment:
rSet.getString("column_name") instead of the index?

Reply:
Fixed